### PR TITLE
Better collisions

### PR DESCRIPTION
**Pull request by Crydsch**
This PR solves issue #1781

### CODE REVIEW

File: examples/models/models_mesh_picking.c [hunk 1]
```diff
@@ -92,30 +100,37 @@ int main(void)
...
-            hitTriangle = true;
         }
-        else hitTriangle = false;
-
-        RayCollision meshHitInfo = { 0 };
+        
+        // Check ray collision against test sphere
+        RayCollision sphereHitInfo = GetRayCollisionSphere(ray, sp, sr);
+        
+        if ((sphereHitInfo.hit) && (sphereHitInfo.distance < collision.distance)) {
```
[the remark this answers is not in the document]
Comment (Crydsch) [reply examples/models/models_mesh_picking.c:108]: Yes, sorry, will do.

**Comment by raysan5** [comment]
@Crydsch I'm merging and I'll review formatting while reviewing the PR.

**Comment by Crydsch** [comment]
Thank you, i haven't had the time to do it yet.

**Comment by raysan5** [comment]
@Crydsch no problem!

**Pull request merged**